### ISSUE TITLE
Check whether `FactoryData` and subclasses can be eliminated

### ISSUE THREAD

**Issue by sebastian-peter** [ie3-institute/PowerSystemDataModel#976]
As another successive issue of #969, there might be a way to eliminate `FactoryData` and all its subclasses (`EntityData` etc.) by creating `AssetInput`s and similar entities directly. 
To make this work, the new constructors in the form of `AssetInputEntityData(EntityData entityData, OperatorInput operator)` (but of course adapted for `AssetInput`s) could be used to iteratively create Inputs from the simplest `UniqueEntity` to the desired input type.

A side effect of this would be that error messages would be easier to create, since there is no way of ensuring that a `FactoryData` also includes UUID and id.

**Comment by staudtMarius** [comment]
@sebastian-peter To do this we'd have to change classes like `AssetInput` and `UniqueInput` from being abstract. This should be done after #1044.

**Comment by sebastian-peter** [comment]
> @sebastian-peter To do this we'd have to change classes like `AssetInput` and `UniqueInput` from being abstract. This should be done after #1044.

Yes, and I'm not really sure if that'd be smart. We should think this through beforehand.

**Comment by staudtMarius** [comment]
@sebastian-peter If you want, we could talk about this next week.
